src/fatlabel.c

```diff
@@ -267,8 +265,8 @@ int main(int argc, char *argv[])
 	}
     }
 
-    if (!set_dos_codepage(-1))	/* set default codepage if none was given in command line */
-        exit(1);
+    if (!set_dos_codepage(codepage))
+        codepage == -1 ? exit(1) : usage(argv[0], 1, 0);
```

Comment:
Ternary operator here (when both functions are void and noreturn) looks very ugly. I think it does not bring any value to call usage() here as user already supplied parseable codepage number which operating system does not support (= this is not a bug nor error in dosfstools) and so showing usage does not help. I would suggest to just unconditionally call exit().

Reply:
I have adjusted my changes accordingly. It also sounds much better with your reasoning. I did not adjust the observable results of the programs to make sure that everything works as before. But I prefer this idea as well.